go/vt/sqlparser/token.go

```diff
@@ -403,11 +403,13 @@ func (tkn *Tokenizer) Scan() (int, []byte) {
 				return int(ch), nil
 			}
 		case '!':
-			if tkn.lastChar == '=' {
+			switch tkn.lastChar {
```

Comment:
can this be simplified not use the switch, since it's basically the same as what it was but with a different return statement?